### PR TITLE
Lets get Spooky 👻

### PR DESCRIPTION
**Pull request by CarlBeek**
Let's actually get some GHOST up in this bitch!

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -233,6 +238,18 @@ def prune(self, new_finalised):
         self.delete_non_subtree(new_finalised, self.root)
         self.root = new_finalised
 
+    def find_head(self):
```
Comment (hwwhww) [inline main.py]:
```suggestion
    def find_head(self) -> Node:
```

File: main.py [hunk 2]
```diff
@@ -233,6 +238,18 @@ def prune(self, new_finalised):
...
+        # calculate the scores of every node starting at the leaves
+        for height in sorted(self.nodes_at_height, reverse=True):
+            for node in self.nodes_at_height[height]:
+                node.score = sum([child.score for child in node.children])
```
Comment (hwwhww) [inline main.py]:
```suggestion
                node.score = sum(child.score for child in node.children)
```

File: main.py [hunk 3]
```diff
@@ -368,6 +385,26 @@ def test_new_finalised_node_pruning():
...
+        _ = tree.add_new_latest_block(block, i)
+
+    val_0_block = tree.latest_block_nodes[0].block
+    for i in range(3):
+        block = Block(val_0_block, 1)
+        _ = tree.add_new_latest_block(block, i)
+
+    val_0_block = tree.latest_block_nodes[0].block
+    # Giving this block more weight, gives GHOST determanism
+    head_node = tree.add_new_latest_block(Block(val_0_block, weight=2), i)
```
Comment (hwwhww) [inline main.py]: by `i`, you mean `2` right?

**Review by hwwhww — APPROVED** [review]

File: main.py (hunk 3 above)
Comment (CarlBeek) [reply main.py]: Yes, that `i` shouldn't be there, well caught. This is the problem with copy pasta.

```suggestion
    head_node = tree.add_new_latest_block(Block(val_0_block, weight=2), 0)
```

**Pull request merged**